src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,64 @@ public VersionParameterDefinition(String repoid, String groupid,
     public VersionParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof StringParameterValue) {
             // TODO: StringParameterValue value = (StringParameterValue) defaultValue;
-            return new VersionParameterDefinition(getRepoid(), "",
-                    "", "", getDescription());
+            return new VersionParameterDefinition(getRepoid(), getGroupid(), getArtifactid(), "", getDescription(),true);
         } else {
             return this;
         }
     }
 
+    public boolean reverseOrder() {
+    	return reverseOrder;
+    }
+    
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if(repoid == null || "ALL".equals(repoid) || repoid.equals(e.getKey())) {
+    			out.add(e.getValue());
+    		}
+    	}
+    	return out;
+    }
+    
+    
     @Exported
     public List<String> getChoices() {
-        Repository r = DESCRIPTOR.getRepo(repoid);
-        List<String> versionStrings = new ArrayList<String>();
-        if (r != null) {
-            File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
-            Aether aether = new Aether(DESCRIPTOR.getRepos(), localRepo);
-            try {
-                List<Version> versions = aether.resolveVersions(groupid, artifactid);
-                for (Version version : versions) {
-                    versionStrings.add(version.toString());
-                }
-            } catch (VersionRangeResolutionException ex) {
-                log.log(Level.SEVERE, "Could not determine versions", ex);
-            }
-            if (!versionStrings.isEmpty()) {
-                // reverseorder to have the latest versions on top of the list
-                Collections.reverse(versionStrings);
-                // add the default parameters
-                versionStrings.add(0, "LATEST");
-                versionStrings.add(0, "RELEASE");
+    	Collection<Repository> repos = getRepos(repoid);
+
+    	List<String> versionStrings = new ArrayList<String>();
+
+    	File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
+        LOG.info("VersionParameterDefinition: local repo "+localRepo.getAbsolutePath());
+    	LOG.log(Level.INFO, "getting versions for: "+groupid+" / "+artifactid);
+    	LOG.log(Level.INFO, " - checking "+repos.size()+" repos" );
+        for(Repository r : repos) {
+        	LOG.log(Level.INFO, " --- checking repo: "+r.getId()+" / "+r.getUrl()+" (user: "+r.getUser()+")" );
+        }
```

Comment:
please only execute this logging block when logging level `INFO` is enabled for this logger